test/performance/alphabet/alphabet_complement_benchmark.cpp

```diff
@@ -0,0 +1,160 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2022, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2022, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/* Copied and adjusted from https://raw.githubusercontent.com/kloetzl/libdna/master/bench2/revcomp.cxx
+ * Credits go to Fabian Klötzl (@kloetzl - https://github.com/kloetzl)
+ */
+
+#include <benchmark/benchmark.h>
+
+#include <random>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/alphabet/views/char_to.hpp>
+#include <seqan3/alphabet/views/complement.hpp>
+#include <seqan3/test/performance/simd_dna4.hpp>
+
+enum class tag
+{
+    revcomp_dna4_inline,
+    seqan3_dna4,
+    seqan3_dna4_vector,
+    seqan3_dna4_simdifyable,
+    seqan3_dna4_simdifyable_vector
+};
+
+class allocator
+{
+public:
+    allocator() = delete;
+    allocator(allocator const &) = delete;
+    allocator & operator=(allocator const &) = delete;
+    allocator(allocator &&) = default;
+    allocator & operator=(allocator &&) = default;
```

Comment:
Can you delete these?
Their usage will lead to double free of `forward` and `reverse`

Reply:
Why? I get it for the copy ctor/assignment.

Reply:
I believe the default implementation of the move operator looks something like this:

```
allocator & operator=(allocator && other) {
    forward = std::move(other.forward);
    reverse = std::move(other.reverse);
}
```

If I have to allocator `a1` and `a2` and now move one into the other `a1 = std::move(a2)`
the original memory of `a1.forward` and `a1.reverse` is lost and not freed.
But when a1 and a2 go out of scope, they point to the same memory and there will be a double free on the same memory.